subsys/bootloader/provision/provision_flash.c

```diff
@@ -24,6 +24,11 @@ typedef struct {
 	} key_data[1];
 } provision_flash_t;
 
+struct monotonic_counter {
+	u32_t num_counter_slots;
+	u16_t counter_slots[1];
```

Comment:
Is this the idiomatic way of saying, 'this is a list and it will be arbitrary long' ?

Reply:
Is guess so, since its done 4 lines above as well :D